tour_site_scraper.rb

```diff
@@ -2,52 +2,91 @@
+    ticket_summary_button_out = driver.find_element(:css, '#Act_Airline_Out')
+    ticket_summary_button_in = driver.find_element(:css, '#Act_Airline_In')
+    return if ticket_summary_button_out.nil? && ticket_summary_button_in.nil?
+    ticket_summary_button_out.click
+    ticket_summary_button_in.click
+
+    WAIT.until { driver.find_element(css: "#Act_response_out .toggle-btn-company").displayed? }
+    WAIT.until { driver.find_element(css: "#Act_response_out .airline-name").displayed? }
+  rescue Exception => e
+    puts 'Trying to fetch data.. ' + retries.to_s
+    retries += 1
+    sleep(1)   # Wait 1s to load the page properly
+    retry if (retries <= MAX_RETRY)
+    raise "Could not get ticket website information: Please give necessary information to search"
   end
-end
 
-# Find available information and available ticket list elements
-ticket_summary = driver.find_elements(:class, 'airline-name')
-ticket_available_lists = driver.find_elements(:class, 'toggle-btn-company')
+  check_return_tickets_visibility(driver)
 
-# Parse elements to find each companies available ticket and sum to get total available tickets
-total_available_ticket = 0
-!ticket_available_lists.nil? && ticket_available_lists.each do |ticket_count|
-  total_available_ticket += ticket_count.text.delete('^0-9').to_i
-end
+  # Scrap Available Tickets Elements
+  ticket_summary = driver.find_elements(:css, '#Act_response_out .airline-name')
+  ticket_available_lists = driver.find_elements(:css, '#Act_response_out .toggle-btn-company')
+
+  # Parse elements to find each companies available ticket and sum
+  total_available_ticket = 0
+  ticket_available_lists&.each do |ticket_count|
+    total_available_ticket += ticket_count.text.delete('^0-9').to_i
+  end
+
+  # Scrap Returning Tickets Elements
+  ticket_summary_in = driver.find_elements(:css, '#Act_response_in .airline-name')
+  ticket_available_lists_in = driver.find_elements(:css, '#Act_response_in .toggle-btn-company')
+
+  # Parse elements to find each companies returning tickets and sum
+  total_available_ticket_in = 0
+  ticket_available_lists_in&.each do |ticket_count_in|
+    total_available_ticket_in += ticket_count_in.text.delete('^0-9').to_i
+  end
+
+  # Write all tickets search results
+  puts 'Total available ticket OUT found is = ' + total_available_ticket.to_s
+  puts 'Total available ticket IN found is = ' + total_available_ticket_in.to_s
 
-puts 'Available ticket companies name = '
-!ticket_summary.nil? && ticket_summary.each do |ticket_cmpany|
-  puts ticket_cmpany.text.to_s + ', ' 
+  puts 'Available ticket IN companies name : '
+  puts '------------------------------------'
+  ticket_summary_in&.each do |ticket_cmpany_in|
+    puts ticket_cmpany_in.text.to_s + ', '
+  end
+
+  puts
+  puts 'Available ticket OUT companies name : '
+  puts '-------------------------------------'
+  ticket_summary&.each do |ticket_cmpany|
+    puts ticket_cmpany.text.to_s + ', '
+  end
 end
```

Comment:
@raselahmed2k7 bhai,  can we plan and organize the code throughout the file using a `class`. It seems to me that few lines of codes are duplicated(line 18, 19, 44, 45) which could be reusable through the use of `method` and the operation + output of this scrapping could be shown through the use of `instance variables`(ticket_summary, ticket_available_lists etc).